app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx

```diff
@@ -156,6 +183,105 @@ export function CategorizedConnections({
     }
   };
 
+  function getConnectionConfig(conn) {
+    const connectionConfig = {
+      name: conn.name,
+      description: conn.description,
+      category: conn.plugin.category,
+      plugin: conn.plugin,
+    };
+
+    return connectionConfig;
+  }
+
+  function exportConnection(conn) {
+    const connectionConfig = getConnectionConfig(conn);
+    const artifact = objectQuery(conn, 'plugin', 'artifact') || {};
+    DownloadFile(
+      connectionConfig,
+      null,
+      `${conn.name}-connector-${artifact.name}-${artifact.version}`
+    );
+  }
+
+  function cloneConnection(conn) {
+    const config = getConnectionConfig(conn);
+    delete config.name;
+
+    openConnectionConfig(config);
+  }
+
+  function openConnectionConfig(config) {
```

Comment:
Nit: With this many handlers, I would prefer to move the state management into a useReducer, to isolate local state updates from actions taken in response to state. I don't it's worth holding up to refactor, perhaps more of a comment for next large component.